docs/docs/solidity_equivalents/inputs.mdx

```diff
@@ -0,0 +1,180 @@
+
+## Comparision of inputs
+
+This section provides a detailed comparison of input data types used in smart contracts, including examples to demonstrate the differences.
+
+The goal is to give a comprehensive understanding of the variations in input parameters required for contract functions
+
+It shows how solidity types in solidity contracts transforms to a cairo type in StarkNet contracts after **warping**.
+
+### Value types
+
+The following types are also called value types because variables of these types will always be passed by value, i.e. they are always copied when they are used as function arguments or in assignments.
+
+#### Booleans
+
+`bool`: The possible values are constants `true` and `false`, these are represenrted in a single `felt` in cairo.
+0 for `false` and 1 for `true`.
+
+#### Integers
+
+Signed and unsigned integers of various sizes (from 8 bits to 256 bits). All integers are represented in a single `felt` with an expection for 256 width integers. 256 bits integers are represented in a pair of `felt`s . Negative Integers are represented as 2-complement of negative of the number.
+
+**Example**: `-1(int256)` is represnted as `Uint256(0xffffffffffffffffffffffffffffffff, 0xffffffffffffffffffffffffffffffff)` in cairo. It should be passed as `0xffffffffffffffffffffffffffffffff, 0xffffffffffffffffffffffffffffffff`.
+
+#### Addresses/Contract Types/Address Literals
+
+Addresses are represented in a single `felt`.
+
+#### Fixed-size byte arrays
+
+The value types bytes1, bytes2, bytes3, …, bytes32 hold a sequence of bytes from one to up to 32. Variables with width upto 248 bits are represented in a single `felt`. `bytes32` is represented in a pair of `felt`s (Uint256).
+
+#### Enums
+
+Enum variables are converted into a single felt in accord with the position of enum deifnition.
+
+#### User Defined Value Types
+
+A user defined value type allows creating a zero cost abstraction over an elementary value type.
+
+Representation is the same as the elementary value type over at which it has been abstracted.
+
+### Reference types
+
+#### String/String Literals
+
+Strings are represented as a `felt` array with the leading element representing the number of elements in the array, typically represented as `(arr_len: felt, arr:felt*)` in the cairo lang.
+
+**Example**: `'warp'` is represented as `4 119 97 114 112`.
+
+#### bytes
+
+Same as `string`, use byte values in-place instead of acii values of characters.
+
+#### Arrays
+
+Arrays can have a compile-time fixed size, or they can have a dynamic size.
+
+- Fixed sized arrays are represented as tuple of base types with length equals length of the array.
+
+  **Example**:
+
+  `uint[3][2]` is represented as `(Uint256, Uint256, Uint256), (Uint256, Uint256, Uint256))`.
+  So, If you want to pass `[[12, 13, 14], [13, 14, 15]]` then you've to provide `12, 0, 13, 0, 14, 0, 13, 0, 14, 0, 15, 0` to interact with warped contracts.
```

Comment:
I would change _**you've**_ to _**you have**_. It looks really weird to do that when _**have**_ is the main verb and not an auxiliary verb (as in the present perfect tense, for example)